contracts/Rollup.sol

```diff
@@ -247,34 +248,44 @@ contract Rollup is RollupHelpers {
         );
 
         changeStakeAmount(governance.STAKE_AMOUNT());
-        bytes32[] memory genesisCommitments = new bytes32[](1);
-        genesisCommitments[0] = keccak256(
-            abi.encodePacked(genesisStateRoot, ZERO_BYTES32)
+        bytes32 genesisCommitment = keccak256(
+            abi.encode(genesisStateRoot, ZERO_BYTES32)
+        );
+        bytes32 commitmentRoot = keccak256(
+            abi.encode(genesisCommitment, ZERO_BYTES32)
         );
         Types.Batch memory newBatch = Types.Batch({
-            commitmentRoot: merkleUtils.getMerkleRootFromLeaves(
-                genesisCommitments
-            ),
-            committer: msg.sender,
-            finalisesOn: block.number + governance.TIME_TO_FINALISE()
+            commitmentRoot: commitmentRoot,
+            meta: Types.encodeMeta(
+                uint256(Types.Usage.Genesis),
+                1,
+                msg.sender,
+                block.number // genesis finalise instantly
```

Comment:
Headsup here. I let the genesis finalise instantly here because we are unlikely to dispute it.

Reply:
Yeah that sounds reasonable